sdk/storage/azure-storage-file-share/src/main/java/com/azure/storage/file/share/implementation/util/ModelHelper.java

```diff
@@ -73,10 +74,35 @@ public static ShareProperties populateShareProperties(SharePropertiesInternal sh
         properties.setLeaseStatus(sharePropertiesInternal.getLeaseStatus());
         properties.setLeaseState(sharePropertiesInternal.getLeaseState());
         properties.setLeaseDuration(sharePropertiesInternal.getLeaseDuration());
-        properties.setEnabledProtocols(ShareEnabledProtocols.parse(sharePropertiesInternal.getEnabledProtocols()));
+        properties.setProtocols(parseShareProtocols(sharePropertiesInternal.getEnabledProtocols()));
         properties.setRootSquash(sharePropertiesInternal.getRootSquash());
         properties.setMetadata(sharePropertiesInternal.getMetadata());
 
         return properties;
     }
+
+    /**
+     * Parses a {@code String} into a {@code ShareProtocols}. Unrecognized protocols will be ignored.
```

Comment:
Is ignoring safe? Or is this just so that we don't break people in the future?

Reply:
This is following .NET's implementation. I believe it is future proofing for when the service starts sending other things, and this is only meant to be used internally, so I think it's ok.